src/wavesurfer.ts

```diff
@@ -434,6 +434,8 @@ class WaveSurfer extends Player<WaveSurferEvents> {
       }
       const onProgress = (percentage: number) => this.emit('loading', percentage)
       blob = await Fetcher.fetchBlob(url, onProgress, fetchParams)
```

Comment:
Since there's a bit of an overlap with the existing fetchParams abortSignal, I'm thinking maybe we can use the new one for this as well?

Something like:

```suggestion
      blob = await Fetcher.fetchBlob(url, onProgress, {
        ...fetchParams,
        signal: signal ?? fetchParams?.signal
      })
```

Then if you abort it, it will also abort the fetching.